src/dune_pkg/checksum.mli

```diff
@@ -20,3 +20,6 @@ val pp : t -> 'a Pp.t
 val equal : t -> t -> bool
 
 module Map : Map.S with type key = t
+
+(** Returns the hash whose kind is the strongest (e.g. SHA512 is stronger than MD5) *)
+val choose_strongest : t list -> t option
```

Comment:
Did this make it in by accident? Doesn't seem related.

Reply:
It's included deliberately. This function is called in toolchain.ml. It lets us validate the compiler package source archive against the strongest checksum in its opam file. Ideally we'd validate it against all its checksums but our `Fetch.fetch` function doesn't currently support this (support will be added in https://github.com/ocaml/dune/pull/10624).

Reply:
Do we choose the "strongest" checksum for regular packages? Why is it special cased for toolchains?

Reply:
Looks like we just use the first checksum for regular packages. Ok I'll change the toolchains logic to also just use the first with the expectation that it will soon be changed to use all the checksums when https://github.com/ocaml/dune/pull/10624 is merged.